blocks/bricks/base.py

```diff
@@ -765,6 +766,16 @@ def get_dims(self, names):
         """
         return [self.get_dim(name) for name in names]
 
+    def get_unique_path(self):
+        """Returns unique path to this brick in the application graph."""
+        if self.parents:
+            parent = min(self.parents, key=operator.attrgetter('name'))
+            return parent.get_unique_path() + [self]
+        else:
+            return [self]
+
+    def get_full_name(self):
```

Comment:
I would leave this method out, and we can move it to the pickler eventually. The precise string format to use might be better decided by the pickler or whatever consumer of the function, because it could depend on many things (e.g. file names can't have `/` or `:` in it).